src/Umbraco.Core/Services/LongRunningOperationService.cs

```diff
@@ -0,0 +1,250 @@
+using Microsoft.Extensions.Logging;
+using Umbraco.Cms.Core.HostedServices;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Persistence.Repositories;
+using Umbraco.Cms.Core.Scoping;
+using Umbraco.Cms.Core.Services.OperationStatus;
+
+namespace Umbraco.Cms.Core.Services;
+
+/// <inheritdoc />
+internal class LongRunningOperationService : ILongRunningOperationService
+{
+    private readonly IBackgroundTaskQueue _backgroundTaskQueue;
+    private readonly ILongRunningOperationRepository _repository;
+    private readonly ICoreScopeProvider _scopeProvider;
+    private readonly TimeProvider _timeProvider;
+    private readonly ILogger<LongRunningOperationService> _logger;
+
+    private readonly TimeSpan _timeToWaitBetweenBackgroundTaskStatusChecks = TimeSpan.FromSeconds(10);
```

Comment:
Perhaps we should have configuration for this, with these being the defaults if no configuration is provided?